### PR TITLE
New `scatter.filter()` and `scatter.data()` methods

### PR DESCRIPTION
**Pull request by flekschas**
This PR addresses #61 and adds two new methods for filtering and updating the data: `scatter.filter()` and `scatter.data()`

### Filter

**API**:

```py
scatter.filter(point_idxs=[0, 1, 2])
```

**Example**:

```py
x = np.random.rand(500)
y = np.random.rand(500)

s1 = jscatter.Scatter(x=x, y=y)
s2 = jscatter.Scatter(x=x, y=y, zoom_on_filter=True)

def on_selection_change(change):
    s2.filter(change.new)

s1.widget.observe(on_selection_change, names='selection')

ipywidgets.HBox([s1.show(), s2.show()])
```

https://user-images.githubusercontent.com/932103/223148819-04060f54-9dc1-424c-bb3c-851f04d7690d.mp4

### Data

**API**:

```py
scatter.data(data=df)
```

**Example**:

```py
data = np.random.rand(500, 4)
df = pd.DataFrame(data, columns=['mass', 'speed', 'pval', 'group'])
df['group'] = df['group'].map(lambda c: chr(65 + round(c)), na_action=None)

s1 = jscatter.Scatter(data=df, x='mass', y='speed')
s2 = jscatter.Scatter(data=df, x='mass', y='speed')

def on_selection_change(change):
    new_data = df.iloc[change.new] if len(change.new) > 0 else df
    s2.data(new_data)

s1.widget.observe(on_selection_change, names='selection')
```

https://user-images.githubusercontent.com/932103/223148337-f0eecaa5-e467-4078-b604-007ab4d323e1.mp4

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -119,6 +137,22 @@ scatter.selection()
 # => array([0, 42, 1337], dtype=uint32)
 ```
 
+<h3><a name="scatter.filter" href="#scatter.filter">#</a> scatter.<b>filter</b>(<i>point_idxs=Undefined</i>)</h3>
+
+Get or set the filtered points. When filtering down to a set of points, all other points will be hidden from the view.
+
+**Arguments:**
+
+- `point_idxs` is either an array-like list of point indices.
```
Comment (manzt) [inline API.md:146]:
```suggestion
- `point_idxs` is a list or an array-like object of point indices.
```

File: API.md [hunk 2]
```diff
@@ -283,20 +317,32 @@ Get or set the x and y axes.
...
 - `legend` is a Boolean specifying if the legend should be shown or not.
 - `position` is a string specifying the legend position. It must be one of `top`, `left`, `right`, `bottom`, `top-left`, `top-right`, `bottom-left`, `bottom-right`, or `center`.
 - `size` is a string specifying the size of the legend. It must be one of `small`, `medium`, or `large`.
+=======
+- `color` is a string referring to a Matplotlib-compatible color.
+- `initiator` is a Boolean value specifying if the click-based lasso initiator should be enabled or not.
+- `min_delay` is an integer specifying the minimal delay in milliseconds before a new lasso point is stored. Higher values will result in more coarse grain lasso polygons but might be more performant. 
+- `min_dist` is an integer specifying the minimal distance in pixels that the mouse has to move before a new lasso point is stored. Higher values will result in more coarse grain lasso polygons but might be more performant.
+- `on_long_press` is a Boolean value specifying if the lasso should be activated upon a long press.
+>>>>>>> dda0106 (Add filter function)
```
Comment (manzt) [inline API.md:345]: op, looks like a merge conflict..

**Review by manzt — APPROVED** [review]
Looks great! Super excited to have this feature :)

File: API.md (hunk 2 above)
Comment (flekschas) [reply API.md:345]: Oh oops. Good catch! :)

**Pull request merged**